Bottom: 4fd2ea5cef18603b55ba7b0f617c45462cc171f1
Top:    1e531feaf6f3da8e20ee1d044d09c63355a74501
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Configure Docker Daemon


---
diff --git a/roles/docker-common/tasks/configure.yml b/roles/docker-common/tasks/configure.yml
new file mode 100644
index 0000000..c1f016e
--- /dev/null
+++ b/roles/docker-common/tasks/configure.yml
@@ -0,0 +1,25 @@
+---
+
+- name: Create Docker Configuration Directory
+  file:
+    path: /etc/docker
+    state: directory
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+
+- name: Configure Docker Daemon
+  template: 
+    dest: /etc/docker/daemon.json
+    src: daemon.json.j2
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+  register: docker_json
+
+- name: Restart Docker Daemon
+  service:
+    name: docker
+    enabled: yes
+    state: reloaded
+  when: docker_json.changed
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index e955efa..f52391f 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -5,10 +5,12 @@
   when: ansible_os_family == "RedHat"
 
 - import_tasks: ip_addr.yml
+
 - import_tasks: packages.yml
 - import_tasks: sshd_settings.yml
 - import_tasks: mariadb_settings.yml
 
+- import_tasks: configure.yml
 
 
 - name: Check if baculavirt selinux module exists
diff --git a/roles/docker-common/templates/daemon.json.j2 b/roles/docker-common/templates/daemon.json.j2
new file mode 100644
index 0000000..57de739
--- /dev/null
+++ b/roles/docker-common/templates/daemon.json.j2
@@ -0,0 +1,4 @@
+{
+  "log-driver": "json-file",
+  "log-opts": {"max-size": "10m", "max-file": "3"}
+}
